torch/csrc/utils/python_arg_parser.h

```diff
@@ -76,6 +76,51 @@
 #include <string>
 #include <vector>
 
+namespace torch {
+inline bool is_symint_node(py::handle obj) {
+  auto static tp_symn = py::type::of<c10::SymIntNodeImpl>();
+  // TODO: switch this to `isinstance`
+  if (obj.get_type().equal(tp_symn)) {
```

Comment:
I suspect that you needed to write it this way because of the ODR violation. I suspect https://github.com/pytorch/pytorch/pull/82552 fixes this. Try it out in a follow up PR.

Reply:
I actually tried `isinstance` first and that didn't work, so I had to resort to doing this ugly equality check.

Reply:
no that's what I mean, you tried it a while ago, and I've fixed the ODR problem now

Reply:
Yeah sorry I phrased it clumsily. I was just confirming that I did try using `isintance` first before I used the equality check. I'll check if https://github.com/pytorch/pytorch/pull/82552 fixes `isinstance`.